### DISCUSSION TITLE
[QUESTION]: ExternalK8sLabelTag功能是否支持获取pod运行时额外打的label

### DISCUSSION THREAD

**Discussion by donkeywon** [alibaba/ilogtail#862] [Help] [answered]
使用ExternalK8sLabelTag功能获取pod的label时，发现有的label可以获取到值，有的label获取到的值是空字符串
![image](https://github.com/alibaba/ilogtail/assets/18396127/ab09d831-f79c-41dd-9706-72dbaafdc193)

获取到空字符串的label是pod运行时通过请求k8s的api动态打的label，可以获取到值的label是pod启动时就有的label

以下是采集配置的input部分
inputs:
    - ContainerFile: true
      ContainerInfo:
        ExternalK8sLabelTag:
            xxx: instance_name
            xxx: engine
            healthy: healthy
            role: master_or_slave
        IncludeK8sLabel:
            xxx: xxx
        K8sNamespaceRegex: xxx
      FilePattern: xxx.log
      LogPath: /xxx/xxx/xxx
      Type: file_log

**Comment by linrunqi08** [comment]
@donkeywon  是使用类似kubectl label pod POD_NAME KEY1=VALUE1 这样的方式加的label吗

**Comment by yyuuttaaoo** [comment, marked as answer]
目前由于logtail不与apiserver通信因此获取不到，对于这个功能可能需要额外开发一个取label的插件，并在部署时增加相关权限。

**Comment by donkeywon** [comment]
@linrunqi08 看operator的代码是get了整个pod信息，然后对label做了些修改再整个put进去

**Comment by linrunqi08** [comment]
@donkeywon  可以参考上面的回复，现在的实现是拿不到apiserver里的信息的。这个可以开发一个插件来针对这个场景，跟apiserver通信，然后把运行时添加的label信息补充上去。我们插件开发还是比较简单的，如果你有兴趣的话，欢迎参与开发

**Comment by donkeywon** [comment]
@linrunqi08 好的

**Comment by linrunqi08** [comment]
@donkeywon  是否有兴趣参与一下呢？我们会提供支持的。
